Домашнее задание 4

AlekseyFilatov opened the pull request:
Домашнее задание 4

Commit:
Домашнее задание 4

svok commented on hw4/.github/workflows/build.yml:
Директория hw4 неуместна. Все домашние задания относятся к одному проекту.

AlekseyFilatov replied:
внес изменения

svok commented on hw4/docker/Dockerfile-driver-java17.yml:
Докерфайлы не должны иметь расширение yml. Это совсем разные форматы

AlekseyFilatov replied:
поправил

svok commented on hw4/docker/spark-3.3.2/kubernetes/dockerfiles/spark/Dockerfile:
В курсе не планируется применять spark. Рекомендую разделить эти две темы: курс по бэкенду на Котлин и spark.

AlekseyFilatov replied:
да, прорабатываю этот вопрос - пробую scalapy

svok commented on hw4/docum/Название и описание проекта.md:
```diff
@@ -0,0 +1,7 @@
+Название проекта:
+Масштабируемая система для выполнения задач распределенного машинного обучения с использованием Kotlin и Spark.
```
1. ML + Spark возможно, но не эффективно. Рекомендую разделять эти две темы и не смешивать.
2. Машинное обучение будем обсуждать в конце курса. Не думаю что следует использовать его раньше времени.
3. Spark - это большая отдельная тема. С нынешним курсом она не совместима.

В рамках выбора темы, вам следует выбрать не стек, а бизнес-назначение проекта. Если планируете использовать не тот стек, который запланирован на курсе, рискуете не освоить ни то, ни другое.

AlekseyFilatov replied:
поправил

svok commented on hw4/docum/Описание MVP.md:
```diff
@@ -0,0 +1,4 @@
+Описание MVP:
+
+Рест сервис забирающий данные из источника(файл) формирует и сохряняет данные для анализа в Cassandra. 
```
Бизнес-назначение так и не описано.
Осторожно. Если делаете неизвестно что, то неизвестно что и получите.

AlekseyFilatov replied:
изучаю

svok commented on hw4/docum/Описание сущностей приложения.md:
```diff
@@ -0,0 +1,20 @@
+ModelEntity
```
Так и непонятно что вы хотите описать этой сущностью. Бизнес-цель не сформулирована, отсюда непонятно что вы хотите этой сущностью описать. 
Еще раз повторю: вряд ли у вас получится затащить на этот курс Spark.

svok commented on hw4/docum/Описание сущностей приложения.md:
```diff
@@ -0,0 +1,20 @@
+ - task_id - номер задания
+ - parameters - параметры задания тренировки модели
+
+ModelEntity - готовая модель хранящаяся в Postgres 
```
1. Здесь нужно сосредоточиться не на том где она будет храниться, а на том какую бизнес-ценность она несет.
2. ML-модели точно не стоит хранить в Постгрессе. Некоторые модели занимают сотни гигабайт места. Постгресс такое не позволит.

AlekseyFilatov replied:
Пока кроме Postgress другие варианты не пробовал

svok commented on hw4/docum/Пользовательские истории.md:
```diff
@@ -0,0 +1,9 @@
+Клиент формирующй ML модель:
+
+User Story : Заходит на платформу - я хочу разместить свои данные на платформе чтобы получить по ним ML модель.
```
Нужна конкретика. Кто ваш пользователь (программист, админ, бухгалтер)? Что он хочет получить? Конкретно в чем задача?

AlekseyFilatov replied:
В данном случае имеется ввиду клиент предоставляющий обучающие данные который хочет получить объяснимую модель

AlekseyFilatov replied:
Поправил документ

svok commented on hw4/docum/Функциональные требования.md:
```diff
@@ -0,0 +1,64 @@
+2.2. Поиск и фильтрация моделей:
+
+Источник требования: Клиенты сервиса.
+Описание требования: Обеспечить возможность поиска и фильтрации моделей по различным параметрам, таким как дата создания, параметры, и другие.
```
Я обычно слово "данные" запрещаю. Потому что это неконкретно. О каких данных вы здесь пишете? Об объеме продаж? Параметры тоже неконкретно. Что за модели? Это тоже неконкретно.

AlekseyFilatov replied:
в данном случае имеется ввиду объяснимая модель

svok commented on hw4/docum/Целевая аудитория и портреты клиентов.md:
```diff
@@ -0,0 +1,33 @@
+Менеджеры:
+
+Категорийный менеджер ежедневно занимающийся управлением ассортимента, промоакциями, ценообразованием, прогнозированием спроса, управлением логистикой.
+Оптимизация на доли процентов даже одного такого бизнес-процесса в масштабе торговой сети приобретает существенный эффект.
```
Самая большая ошибка - это делать универсальный инструмент еще до того как сделано хоть одно конкретное решение. Если ваша ЦА - это вся планета Земля и ее окресности, то такой стартап точно не взлетит. Хотя бы потому, что это очень дорого предусмотреть все возможности.

AlekseyFilatov replied:
здесь добавил общий пример из описаний применения моделей машинного обучения - пока не конкретно - разбираюсь

svok commented on hw4/specs/specs-ml-v1.yaml:
```diff
@@ -0,0 +1,851 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MlReadResponse'
+        400:
```
В боевом проекте возможны сотни или даже тысячи разных ошибок. Вы точно хотите их все описывать в API? Вы точно готовы пересобирать все связанные проекты после добавления очередной ошибки?

AlekseyFilatov replied:
удалил варианты ошибок

svok reviewed (commented):
Алексей, прошу полностью убрать все директории типа hw1-hw4. Мы не сможем проверять ДЗ, когда вы весь проект копируете с нуля.

Commit:
Домашнее задание 4 - Работа над ошибками :
- Модификация проекта - удалены каталоги hw1-4, добавлен ml-project

Commit:
Домашнее задание 4 - Работа над ошибками :
- Модификация проекта - измена спецификация для генерации моделей и пользовательские истории

svok commented on ml-project/lessons/.gradle/8.12.1/executionHistory/executionHistory.lock:
Все содержимое директории .gradle является мусором. Его нужно заигнорить в .gitignore

svok commented on ml-project/ml-service/ml-service-api-v1-mappers/src/main/kotlin/MappersV1FromTransport.kt:
```diff
@@ -0,0 +1,104 @@
+import api.kotlinproject.mappers.v1.exceptions.UnknownRequestClass
+
+
+fun MkplContext.fromTransport(request: IRequest) = when (request) {
```
Префикс Mkpl у вас неуместен во всем проекте

svok reviewed (commented):
Алексей, хорошая работа. Обратите внимание на комментарии.
